### DISCUSSION TITLE
When is Codeception 5 going to be released?

### DISCUSSION THREAD

**Discussion by 99hops** [Codeception/Codeception#6327] [Q&A]
I googled around for Codeception roadmap or any release date plan related to 5 release  but found nothing. 
So does anyone have an idea when we could expect this to happen? :smiley:

**Comment by brainexe** [comment]
I was just looking for the same information, as it's a blocker to upgrade our project to Symfony 6 right now.

**Comment by 99hops** [comment]
This is exactly why I am asking about it

**Comment by Naktibalda** [comment]
It was planned to be released after release of PHPUnit 10 (Originally in February 2021), but PHPUnit is almost a year late.

I think that Symfony 6 is good enough reason for release of Codeception 5, so we can stop waiting for PHPUnit 10 and release it as soon as possible.

**Comment by codedge** [comment]
Just for a reference for everybody waiting for the release, see https://github.com/Codeception/module-symfony/issues/152#issuecomment-1014469033

In short:
Codeception still waits for PHPUnit 10 to be released.

**Comment by Naktibalda** [comment]
I expect to release 5.0.0-alpha1 next week.

I decoupled test execution and reporting from PHPUnit, but that work isn't finished yet.
If you have any custom code that relies on PHPUnit's TestListener interface, it must be rewritten to use Codeception events instead. (Most likely case is custom reporters).

alpha1 will depend on development branches of PHPUnit and its libraries.

**Comment by aedart** [comment]
Hi

Is there any chance that codeception 5.x could be release using PHPUnit  9.x, instead of waiting for PHPUnit 10.x?
Furthermore, would it be possible to include version 10 as a compatible version, e.g. "phpunit/phpunit": "^9.5 || ^10.0"?
(_Assuming that breaking changes in PHPUnit are somehow handled by codeception_)

**Reply by Naktibalda** [reply]
I released 5.0.0-alpha2 which depends on development branches of PHPUnit too.
You can try to use it.

**Reply by Naktibalda** [reply]
Support for PHPUnit 9 https://github.com/Codeception/Codeception/pull/6380

**Comment by Naktibalda** [comment]
I released 5.0.0-RC1 with support for PHPUnit9. https://github.com/Codeception/Codeception/releases/tag/5.0.0-RC1

Some feedback would be nice.

**Reply by burned42** [reply]
I tried updating to that version but unfortunately composer didn't find a compatible version of `codeception/module-symfony`. I also tried using the 3.0 branch of the symfony module (assuming that it's the preparation for the new release) but that yields the following problem:

```
  Problem 1
    - Root composer.json requires codeception/module-symfony 3.0.x-dev -> satisfiable by codeception/module-symfony[3.0.x-dev].
    - codeception/module-symfony 3.0.x-dev requires codeception/codeception dev-5.0-interfaces as 5.0.x-dev -> found codeception/codeception[dev-replace-hoa-console, dev-shell-improv, dev-udp, dev-collision, dev-issue-6086, dev-4.1-runbefore-error-output, dev-4.1-generate-actions-with-self-type, dev-5.0-fix-code-coverage-build, dev-5.0-extract-constraints, dev-5.0-phpunit9, 1.0.9, ..., 1.8.x-dev, 2.0.0-alpha, ..., 2.5.x-dev, 3.0.0, ..., 3.1.x-dev, 4.0.0, ..., 4.1.x-dev, 5.0.0-alpha1, ..., 5.0.x-dev] but it does not match the constraint.
```

**Reply by Naktibalda** [reply]
I released 3.0.0

**Reply by burned42** [reply]
Thanks, update worked fine now:

```
  Lock file operations: 2 installs, 6 updates, 1 removal
  - Removing codeception/phpunit-wrapper (9.0.7)
  - Upgrading codeception/codeception (4.1.31 => 5.0.0-RC1)
  - Upgrading codeception/lib-innerbrowser (2.0.2 => 3.1.1)
  - Locking codeception/lib-web (1.0.0)
  - Locking codeception/lib-xml (1.0.0)
  - Upgrading codeception/module-asserts (2.0.1 => 3.0.0)
  - Upgrading codeception/module-rest (2.0.1 => 3.1.0)
  - Upgrading codeception/module-symfony (2.1.1 => 3.0.0)
  - Upgrading softcreatr/jsonpath (0.7.5 => 0.8.0)
```

Running the tests also worked fine after the update with one exception:

> [Error] Class "Codeception\Util\Stub" not found 

I needed to replace usages of `Codeception\Util\Stub` with `Codeception\Stub` which doesn't look like it's mentioned in [the changelog](https://github.com/Codeception/Codeception/blob/5.0/CHANGELOG-5.x.md) yet. Since it worked with version 4, I guess that's a BC break worth mentioning.

Coverage report also works and looks just as usual :) so thank you, looks like everything works, even with PHPUnit 9 support now :+1:

**Reply by W0rma** [reply]
Seems like releases compatible with Codeception 5 are missing for the modules [`codeception/module-amqp`](https://github.com/Codeception/module-amqp) and [`module-phpbrowser`](https://github.com/Codeception/module-phpbrowser) as well.

**Comment by sebastianneubert** [comment]
PHPUnit 10 drops support for PHP8.0 and require PHP8.1 minimum. Same as Symfony v6.1

https://github.com/sebastianbergmann/phpunit/commit/95b61586c5d9b90bcaa516ddca0af00efc408940

**Reply by Naktibalda** [reply]
I noticed, it isn't a problem.

**Comment by Naktibalda** [comment]
I released 5.0.0 RC2 - https://github.com/Codeception/Codeception/releases/tag/5.0.0-RC2

**Comment by codedge** [comment]
Any updates here? I know things take time, but Symfony 6 was released in Nov 2021, 8 months ago. Is there any timeframe for Codeception 5 w/ Symfony 6 support or anything to speed up the process? I mean a stable release, no Release Candidate.

**Comment by developedsoftware** [comment]
I would also like to know when 5.0 is going to be released (appreciate the release candidates but would be nice to have a 5.0 release)

**Comment by Naktibalda** [comment]
I released [4.2.0](https://github.com/Codeception/Codeception/releases/tag/4.2.0) today, I will merge these changes to 5.0 branch and release 5.0.0 this weekend or next week, unless someone reports critical issue that needs fixing.

**Comment by developedsoftware** [comment]
Awesome! Thank you!

**Comment by Naktibalda** [comment]
1. Merge https://github.com/Codeception/Codeception/pull/6474
2. Fix https://github.com/Codeception/Codeception/issues/6473 (trivial)
3. Try to fix compatibility with PHPUnit main branch again.

**Comment by Naktibalda** [comment]
I will release another RC version after finishing https://github.com/Codeception/Codeception/pull/6475
If nothing else comes up, stable 5.0.0 can be released a week later.

**Comment by Naktibalda** [comment]
I released 5.0.0-RC4


* Implemented basic attribute support (#6449) by @DavertMik
* Significantly reduced dependencies on PHPUnit
* Replaced PHPUnit\Framework\TestResult with ResultAggregator
* Added assertionCount method to ResultAggregator 
* DotReporter prints standard result summary (#6441) by @Orchestrator404
* Fixed DotReporter output format
* Fixed fetching remote code coverage data (#6448)
* Loading .env file must not override existing environment variables (#6477)
* All changes from 4.2.0 and 4.2.1:
  * Improved multi-application experience, allow filtering suites by name (#6435) by @calvinalkan
  * Configuration override is passed to included suites (#5978) by @calvinalkan
  * Made dry-run command work with module methods having return types (#6470)
  * Support for expectError/Warning/Notice/Deprecation methods in unit tests (Requires PHPUnit 8.4+)
  * Implemented new setting `convert_deprecations_to_exceptions` (#6469)
  * Action file generator: Do not return when return type is never (#6462)
  * Execute setupBeforeClass/tearDownAfterClass only once (#6481)
